examples/tensorflow/yolo_v5_real_time_with_labels.py

```diff
@@ -0,0 +1,152 @@
+    return classes  # return classes (int)
+
+def YOLOdetect(output_data):  # input = interpreter, output is boxes(xyxy), classes, scores
+    output_data = output_data[0]                # x(1, 25200, 7) to x(25200, 7)
+    boxes = np.squeeze(output_data[..., :4])    # boxes  [25200, 4]
+    scores = np.squeeze( output_data[..., 4:5]) # confidences  [25200, 1]
+    classes = classFilter(output_data[..., 5:]) # get classes
+    # Convert nx4 boxes from [x, y, w, h] to [x1, y1, x2, y2] where xy1=top-left, xy2=bottom-right
+    x, y, w, h = boxes[..., 0], boxes[..., 1], boxes[..., 2], boxes[..., 3] #xywh
+    xyxy = [x - w / 2, y - h / 2, x + w / 2, y + h / 2]  # xywh to xyxy   
+
+    return xyxy, classes, scores  # output is boxes(x,y,x,y), classes(int), scores(float) [predictions length]
+
+
+def main():
+    global rectangles
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--model', help='Path of the detection model.', required=True)
+    parser.add_argument('--label', help='Path of the labels file.')
+    args = parser.parse_args()
+    
+    if args.label: 
+        labels = ReadLabelFile(args.label)
+    else:
+        labels = None
+    picam2 = Picamera2()
+    picam2.start_preview(Preview.QTGL)
+    
+    stream_format = "YUV420"
+    if Picamera2.platform == "Platform.PISP":
+        stream_format = "RGB888"
+        
+    config = picam2.create_preview_configuration(main={"size": normalSize},
+                                                 lores={"size": lowresSize, "format": stream_format})
+    picam2.configure(config)
+
+    stride = picam2.stream_configuration("lores")["stride"]
+    picam2.post_callback = DrawRectangles
+        
+    picam2.start()
+    interpreter = tflite.Interpreter(model_path=args.model, num_threads=4)
+    interpreter.allocate_tensors()
+
+    while True:
+        img = picam2.capture_array("lores")
+        
+        input_details = interpreter.get_input_details()
+        output_details = interpreter.get_output_details()
+        height = input_details[0]['shape'][1]
+        width = input_details[0]['shape'][2]
+        
+        floating_model = False
+        if input_details[0]['dtype'] == np.float32:
+            floating_model = True
+        
+        if stream_format == "YUV420":
+            img = cv2.cvtColor(img, cv2.COLOR_YUV420p2RGB)
+            
+        new_shape = (height, width) # the shape the model was trained with
+        img = cv2.resize(img,new_shape)
+        input_data = np.expand_dims(img, axis=0)
+        if floating_model:
+            input_data = (np.float32(input_data) - 127.5) / 127.5
+
+        interpreter.set_tensor(input_details[0]['index'], input_data)
+        interpreter.invoke()
+        output_data = interpreter.get_tensor(output_details[0]['index']) 
+        xyxy, classes, scores = YOLOdetect(output_data) 
+        rectangles = []
+        for i in range(len(scores)):
```

Comment:
I expect Python fans might prefer
```
    for xyxy_item, classes_item, score in zip(xyxy, classes, scores):
```
(Not too bothered though, myself!)

Reply:
I aggree, but if i understand the yolo output format and zip() correctly then I think you need to loop over it like this because the outermost dimension is the 4 seperate points for the bbox, and the innermost is the list of different detections, so in order for this to work I would have to do an extra operation (reshape)